### PR TITLE
Call reject on exception in search.onsuccess

### LINKED ISSUE #2759: database errors are not properly passed around
Report from user:

> Whenever I try to write an email, it keeps loading after I put the email adress. The button to send is not enabled. What is going on?

They were experiencing an exception that was happening probably in backend page:

> Error: BrowserMsg() sendRawResponse::NotFoundError: IDBDatabase.transaction: 'contacts' is not a known object store name

This should have resulted in a thrown exception on the compose page, which should result in an error modal. The exception got lost somewhere on the way, resulting in "endless loading".

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
Closes #2759 

Raising an exception inside 
```
       search.onsuccess = Catch.try(() => {
...
})
```
block resulted in unresolved Promise, so `const raw: unknown[] = await new Promise(...` hang.

### CODE REVIEW

**Comment by rrrooommmaaa** [comment]
or perhaps, we shouldn't pass `reject` to `reportErr`, but make `Catch.try` a bit more complicated.

File: extension/js/common/platform/store/contact-store.ts [hunk 1]
```diff
@@ -297,7 +297,7 @@ export class ContactStore extends AbstractStore {
             cursor.continue();
           }
         }
-      });
+      }, reject);
```
Comment (tomholub) [inline extension/js/common/platform/store/contact-store.ts:300]: I think better just stop using Catch.try in this situation. Instead put our own try/catch block inside the callback method here. Something like

```ts
search.onsuccess = () => {
  try {
    ...
  } catch(e) {
    reject(e)
  }
}
```

Comment (tomholub) [reply extension/js/common/platform/store/contact-store.ts:300]: Then Catch.try can stay as is.

Good find btw.

**Review by tomholub — CHANGES_REQUESTED** [review]
This looks good - assuming you have tested it to properly pass the error back to the caller.

Can you please do the same with all of the `search.onsuccess = ` in other methods? It seems the existing `Catch.try` will be swallowing up the errors (and reporting them) leaving the caller hanging.

**Review by tomholub — APPROVED** [review]

**Pull request merged**